### ISSUE TITLE
Integration test for node selection

### ISSUE THREAD

**Issue by deyanzz** [hashgraph/hedera-sdk-cpp#167]
New integration tests should be added to test the node selection for:

- `Node`
- `Client`
- `Executable`

These tests should be executed against a working [Hedera Local Node](https://github.com/hashgraph/hedera-local-node). All necessary configurations should be done also.

To fully test the backoff time when connecting to a failed node, the following task should be completed first for the Local Node:

- https://github.com/hashgraph/hedera-local-node/issues/120
- https://github.com/hashgraph/hedera-local-node/issues/309

**Comment by deyanzz** [comment]
Currently I'm working on https://github.com/hashgraph/hedera-sdk-cpp/issues/201 as it's a blocker for https://github.com/hashgraph/hedera-sdk-cpp/issues/164. This issue is set `On Hold`.

**Comment by deyanzz** [comment]
Moving back to this issue, as the following task for [Hedera Local Node](https://github.com/hashgraph/hedera-local-node) have been completed:
- https://github.com/hashgraph/hedera-local-node/issues/120
- https://github.com/hashgraph/hedera-local-node/issues/309
